sympy/functions/elementary/tests/test_exponential.py

```diff
@@ -383,6 +384,18 @@ def test_log_symbolic():
     assert log(-p).func is log and log(-p).args[0] == -p
 
 
+def test_special_exp():
+    for d in tuple(cos_table()) + (4, 6, 8, 10, 12):
+        # use 13 to affirm numerator does not need to be 1
+        d/=S(13)
+        assert d != 1
+        ans = cos(pi/d) + I*sin(pi/d)
+        assert not ans.has(cos)
+        assert exp(I*pi/d) == ans
+        d = -d
+        assert exp(I*pi/d) == cos(pi/d) + I*sin(pi/d)
```

Comment:
I don't think that this is a good transformation. It turns something that is not an Add into an Add. Complex exponentials canonicalise better under multiplication.

With master:
```
In [1]: d = exp(13*I*pi/12)

In [2]: d * d.conjugate()
Out[2]: 1
```
With the PR:
```
In [2]: d = exp(13*I*pi/12)

In [3]: d * d.conjugate()
Out[3]:
⎛  √6   √2     ⎛  √6   √2⎞⎞ ⎛  √6   √2     ⎛  √6   √2⎞⎞
⎜- ── - ── - ⅈ⋅⎜- ── + ──⎟⎟⋅⎜- ── - ── + ⅈ⋅⎜- ── + ──⎟⎟
⎝  4    4      ⎝  4    4 ⎠⎠ ⎝  4    4      ⎝  4    4 ⎠⎠
```

Reply:
I agree. Will close the issue with recommendation to use `rewrite(cos)` when desired.